File: 1431. Kids With the Greatest Number of Candies/CnE.cpp
class Solution
{
public:
    vector<bool> kidsWithCandies(vector<int> &candies, int extraCandies)
    {
        using namespace std;
        int maxCandies = *max_element(candies.begin(), candies.end());
        // max_element returns an iterator, use * for deferencing to get the element (max value)
        vector<bool> res;
        res.reserve(candies.size()); 
        // Reserve space to avoid multiple allocations, improve performance
        for (int c : candies) res.push_back(c + extraCandies >= maxCandies);
        return res;
    }
};

File: 1431. Kids With the Greatest Number of Candies/CnE.js
/**
 * @param {number[]} candies
 * @param {number} extraCandies
 * @return {boolean[]}
 */
var kidsWithCandies = (candies, extraCandies) => {
    return candies.map(c => c + extraCandies >= Math.max(...candies));
}